fe/fe-core/src/main/java/com/starrocks/statistic/base/ColumnClassifier.java

```diff
@@ -0,0 +1,102 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.statistic.base;
+
+import com.google.common.collect.Lists;
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.StructType;
+import com.starrocks.catalog.Table;
+import com.starrocks.catalog.Type;
+
+import java.util.List;
+
+/*
+ * The classification for the statistics column is split
+ * into PrimaryType Column/ComplexType Column/Subfield Column
+ */
+public class ColumnClassifier {
+
+    private final List<ColumnStats> columnStats = Lists.newArrayList();
+
+    private final List<ColumnStats> unSupportStats = Lists.newArrayList();
+
+    public static ColumnClassifier of(List<String> columnNames, List<Type> columnTypes, Table table) {
+        ColumnClassifier columnClassifier = new ColumnClassifier();
+        columnClassifier.classifyColumnStats(columnNames, columnTypes, table);
+        return columnClassifier;
+    }
+
+    private void classifyColumnStats(List<String> columnNames, List<Type> columnTypes, Table table) {
+        for (int i = 0; i < columnNames.size(); i++) {
+            String columnName = columnNames.get(i);
+            Type columnType = columnTypes.get(i);
+
+            if (table.getColumn(columnName) != null) {
+                if (columnType.canStatistic()) {
+                    columnStats.add(new PrimitiveTypeColumnStats(columnName, columnType));
+                } else {
+                    unSupportStats.add(new ComplexTypeColumnStats(columnName, columnType));
+                }
+            } else {
+                int start = 0;
+                int end;
+                List<String> names = Lists.newArrayList();
+                while ((end = columnName.indexOf(".", start)) > 0) {
+                    start = end + 1;
+                    String name = columnName.substring(0, end);
+                    Column c = table.getColumn(name);
+                    if (c != null && c.getType().isStructType()) {
+                        names.add(name);
+                        columnName = columnName.substring(end + 1);
+                        Type type = c.getType();
+                        if (!columnName.contains(".")) {
+                            names.add(columnName);
+                        } else {
+                            int subStart = 0;
+                            int pos = 0;
+                            int subEnd;
+                            while ((subEnd = columnName.indexOf(".", pos)) > 0 && type.isStructType()) {
```

Comment:
this piece of code is too complex to read

Reply:
to split&vaild struct column, like `a.b.c.d`